specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json

```
@@ -1359,9 +1361,249 @@
+              "x-ms-secret": true,
+              "type": "string"
+            }
+          },
+          "required": [
+            "username",
+            "password"
+          ]
+        }
+      },
+      "type": "object",
+      "required": [
+        "primaryDnsIpAddress",
+        "secondaryDnsIpAddress",
+        "domainName",
+        "smbServerName"
+      ]
+    },
+    "CacheUsernameDownloadSettings": {
+      "description": "Settings for username and group download for Extended groups.",
+      "type": "object",
+      "properties": {
+        "extendedGroups": {
+          "type": "boolean",
+          "description": "This indicates if Extended Groups is enabled."
+        },
+        "usernameSource": {
+          "description": "This setting determines how the system gets username and group names for clients.",
+          "enum": [
+            "AD",
+            "LDAP",
+            "File",
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "UsernameSource",
+            "modelAsString": true
+          },
+          "type": "string",
+          "default": "None"
+        },
+        "groupFileURI": {
+          "type": "string",
+          "description": "The URI of the file containing the group (etc/group) information. This field must be populated when 'usernameSource' is set to 'File'."
+        },
+        "userFileURI": {
+          "type": "string",
+          "description": "The URI of the file containing the user (etc/passwd) information. This field must be populated when 'usernameSource' is set to 'File'."
+        },
+        "ldapServer": {
+          "type": "string",
+          "description": "The fully qualified domain name or IP address of the LDAP server to use."
+        },
+        "ldapBaseDn": {
+          "type": "string",
+          "description": "The base distinguished name for the LDAP domain."
+        },
+        "encryptLdapConnection": {
+          "type": "boolean",
+          "description": "This indicates if the LDAP connection should be encrypted."
+        },
+        "requireValidCertificate": {
           "type": "boolean",
-          "description": "root squash of cache property."
+          "description": "Determines if the certificates should be validated by a certificate authority. When true caCertificateURI must be provided."
+        },
+        "autoDownloadCertificate": {
+          "type": "boolean",
+          "description": "Determines if the certificate should be automatically downloaded. This applies to 'caCertificateURI' when 'requireValidCertificate' is true, or a self signed certificate otherwise."
+        },
+        "caCertificateURI": {
+          "type": "string",
+          "description": "The URI of the CA certificate to validate the LDAP secure connection. This field must be populated when 'requireValidCertificate' is set to 'true'."
+        },
+        "usernamePolled": {
```

Minor: Maybe rename to `usernameDownloaded`? Or change the description to reference polling (though the request field refers to downloading and no other relevant "polling" references exist).

---

we can make this change

---

Cool, thanks! I know the RP and API will have to be aligned on this.

@brpanask , git-suggestion for the field name.

@mannywarrior , I'll open a work item to change the field name and assign it over.

```suggestion
        "usernameDownloaded": {
```

---

Fixed in next commit.